### PR TITLE
Add grouping functionality to OcTableFiles

### PR DESCRIPTION
**Pull request by elizavetaRa**
## Description
Grouping as well as preview functionality for the tables. 

Setup is done with the property "groupingSettings" . Following settings are possible:
- groupingFunctions: Object with keys as grouping options names and functions that get a table data row and return a group name for that row. The names of the functions are used as grouping options. 

- groupingBy: must be either one of the keys in groupingFunctions or 'None'. If not set, default grouping will be 'None'.
- ShowGroupingOptions: boolean value for showing or hinding the select element with grouping options above the table.
- PreviewAmount: Integer value that is used to show only the first n data rows of the table.object

Screenshot from 2021-09-08 10-47-39

## Motivation and Context
UX-friendly option to make the table more

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:

- Tests

### CODE REVIEW

**Comment by labkode** [comment]
Plese merge is green before we get into conflicts

File: src/components/table/OcTableFiles.vue [hunk 1]
```diff
@@ -140,6 +141,17 @@ export default {
...
   },
   props: {
+    /**
+     * Grouping settings for the table. Following settings are possible:<br />
+     * -**groupingFunctions**: Object with keys as grouping options names and functions that get a table data row and return a group name for that row. The names of the functions are used as grouping options.
+     * -**groupingBy**: must be either one of the keys in groupingFunctions or 'None'. If not set, default grouping will be 'None'.<br />
+     * -**ShowGroupingOptions**:  boolean value for showing or hinding the select element with grouping options above the table. <br />
+     * -**PreviewAmount**: Integer value that is used to show only the first n data rows of the table.
+     */
+    groupingSettings: {
```
Comment (kulmann) [inline src/components/table/OcTableFiles.vue:151]: I'd prefer `grouping` as prop name. `groupingSettings` sounds too verbose.

**Review by kulmann — CHANGES_REQUESTED** [review]
This PR does two things, please split it into two PRs:
1: different icon variation for files vs folders
2: grouping for files table. BUT: the actual grouping logic for the `OcTable` is completely missing from this PR.

Without having seen the grouping implementation yet, please also:
a: add extensive units tests for your new grouping functionality
b: add custom prop validation for the grouping prop

I can see line break changes in you PR - are they necessary? Would be good to reduce that kind of noise in PRs.

Lastly, please accept the CLA, otherwise we can't merge your contribution.

Please get in touch with me if you need help in any way. Happy to assist!

File: src/components/table/OcTableFiles.vue [hunk 2]
```diff
@@ -65,9 +66,9 @@
       <oc-resource-size :size="item.size" />
     </template>
     <template #owner="{ item }">
-      <oc-avatars
```
Comment (pascalwengerter) [inline src/components/table/OcTableFiles.vue:68]: This looks like you haven't properly rebased your PR - the `oc-avatar-group` has been replaced by a (new) `oc-avatars` component which expects an `items` prop and not the (old) `users`

File: src/components/table/OcTableFiles.vue [hunk 3]
```diff
@@ -679,8 +658,6 @@ export default {
   <oc-table-files :resources="resources" :arePathsDisplayed="true" v-model="selected" />
 </template>
 <script>
-  import { shareType } from "../../utils/shareType"
```
Comment (pascalwengerter) [inline src/components/table/OcTableFiles.vue:682]: This also is on (our) current master and looks like you're running an outdated version. Please rebase on our master and apply your changes there

**Review by pascalwengerter — CHANGES_REQUESTED** [review]

**Comment by diocas** [comment]
Let's close this PR as this shouldn't be trying to merge our QA branch which has other stuff... Let's discuss it tomorrow and do a different PR for these things... @elizavetaRa

**Pull request closed without merging**